gateway-messages/src/mgs_to_sp.rs

```diff
@@ -126,6 +128,22 @@ pub enum MgsRequest {
     },
 
     SerialConsoleKeepAlive,
+
+    /// Reset a specific component
+    /// SP_ITSELF and ROT are supported
+    ResetComponentPrepare {
+        component: SpComponent,
+    },
+    ResetComponentTrigger {
+        component: SpComponent,
+    },
+
+    /// Change boot image selection on reset or power-on.
+    SwitchDefaultImage {
+        component: SpComponent,
+        slot: SlotId,
+        duration: SwitchDuration,
+    },
```

Comment:
Can you expand the `gateway_messages::tests::versioning::v2::mgs_request()` test to cover these new variants? (This is the test that ensures future changes don't break the on-the-wire form of any pre-existing variants.)